### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by zichenglang** [handshake-org/hsd#246]
## Env
- System : macOS Mojave 10.14.6
- NPM : 6.11.2
- Node.js : v12.7.0
- Python : 2.7.10

## Step
Follow the instructions exactly
```
$ git clone git://github.com/handshake-org/hsd.git
$ cd hsd
$ npm install --production
```
## Error
When I run command `npm install --production`, I got this

```log
  CXX(target) Release/obj.target/bcrypto/src/aead.o
../src/aead.cc:39:68: error: too few arguments to function call, single argument 'context' was not specified
  target->Set(Nan::New("AEAD").ToLocalChecked(), ctor->GetFunction());
                                                 ~~~~~~~~~~~~~~~~~ ^
/Users/lacdon/Library/Caches/node-gyp/12.7.0/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/lacdon/Library/Caches/node-gyp/12.7.0/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
1 error generated.
make: *** [Release/obj.target/bcrypto/src/aead.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.7.0
gyp ERR! command "/usr/local/Cellar/node/12.7.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/lacdon/github/hsd/node_modules/bcrypto
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
Build failed.

> mrmr@0.1.8 install /Users/lacdon/github/hsd/node_modules/mrmr
> node-gyp rebuild || echo 'Build failed.'

  CXX(target) Release/obj.target/mrmr/src/murmur3.o
  CXX(target) Release/obj.target/mrmr/src/mrmr.o
  SOLINK_MODULE(target) Release/mrmr.node

> unbound@0.3.4 install /Users/lacdon/github/hsd/node_modules/unbound
> node-gyp rebuild

  CXX(target) Release/obj.target/unbound/src/node_unbound.o
../src/node_unbound.cc:4:10: fatal error: 'unbound.h' file not found
#include <unbound.h>
         ^~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/unbound/src/node_unbound.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.7.0
gyp ERR! command "/usr/local/Cellar/node/12.7.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/lacdon/github/hsd/node_modules/unbound
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
```
and this

```
4 warnings generated.
  CC(target) Release/obj.target/goosig/src/goo/hmac.o
  CC(target) Release/obj.target/goosig/src/goo/random.o
  CXX(target) Release/obj.target/goosig/src/goosig.o
../src/goosig.cc:47:67: error: too few arguments to function call, single argument 'context' was not specified
  target->Set(Nan::New("Goo").ToLocalChecked(), ctor->GetFunction());
                                                ~~~~~~~~~~~~~~~~~ ^
/Users/lacdon/Library/Caches/node-gyp/12.7.0/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/lacdon/Library/Caches/node-gyp/12.7.0/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
1 error generated.
make: *** [Release/obj.target/goosig/src/goosig.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.7.0
gyp ERR! command "/usr/local/Cellar/node/12.7.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/lacdon/github/hsd/node_modules/goosig
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
Build failed.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: unbound@0.3.4 (node_modules/unbound):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: unbound@0.3.4 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

added 36 packages from 9 contributors and audited 124 packages in 54.794s
found 0 vulnerabilities
```
I got an error message when I try to run hsd
```
./bin/hsd
Bindings for bcrypto were not built.
Please build them before continuing.
```

**Comment by zichenglang** [comment]
## Fix 
It maybe fix this problem
```
npm install bcrypto https://github.com/cryptocoinjs/secp256k1-node
```
There are still some error when install bcrypto
```
  CXX(target) Release/obj.target/unbound/src/node_unbound.o
../src/node_unbound.cc:4:10: fatal error: 'unbound.h' file not found
#include <unbound.h>
         ^~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/unbound/src/node_unbound.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.7.0
gyp ERR! command "/usr/local/Cellar/node/12.7.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/lacdon/github/hsd/node_modules/unbound
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
```
So I try to reinstall unbound 
```
 brew install unbound
```
The problem still there, but the `hsd` can run. I'm really confused

**Comment by chjj** [comment]
I think the current version of bcrypto hsd is pointing to is out of date. Will update.

> ../src/node_unbound.cc:4:10: fatal error: 'unbound.h' file not found

Unbound is optional. It's really only useful if you plan on running a public recursive resolver. The javascript resolver is a bit slower but probably sufficient for local use.

Though, this makes me wonder, does the brew package for unbound include header files?

**Comment by boymanjor** [comment]
@chjj I noticed hsd is still pointing to bcrypto version ~3.1.0. I can update this to the latest published npm package.

re: the unbound brew package - I have a brew installation of unbound on my OS X machine, and it includes the header file. I'm not familiar with the process for creating brew packages (formulae), but this [link](https://github.com/Homebrew/homebrew-core/blob/95aaba89a9d4e3ef708d79e31b78961b6a0765a7/Formula/unbound.rb) seems to indicate the package manager is simply pulling the source files onto the machine, configuring, building, and then installing unbound.

I have included a screenshot of the result of `brew list unbound`. It should be noted that I am using an older version of macOS (v10.13.6 ) as well as an older version of the unbound package (v1.9.1).

![brew list unbound](https://user-images.githubusercontent.com/1259462/65531854-d4693180-dec8-11e9-8ebc-094908c17b21.png)
